rse-advanced-git.Rmd

```diff
@@ -0,0 +1,945 @@
+(from the web address line or click 'Clone and download' and copy that).
+
+*TODO: Figure - Clone Origin [../fig/github_screenshot_cloneOrigin.png](https://github.com/UW-Madison-DataScience/git-novice-custom/tree/gh-pages/fig/github_screenshot_cloneOrigin.png)*
+
+In terminal or Gitbash, navigate to a folder you'd like to hold this repo,
+we will place it on our `Desktop`.
+Once there you can use the `clone` command with the link you copied as the first argument.
+
+```shell
+$ cd ~/Desktop
+$ git clone https://github.com/USERNAME/countries.git
+```
+
+> **Why does the command above say 'USERNAME'?**
+>
+> So that we can't copy the command above and accidentally clone someone else's
+> version of countries to our computer, the command above uses the placeholder
+> 'USERNAME' where you should put your own username if your copied from above
+> instead of copying the link from your browser and pasting it into the command.
+
+```text
+Cloning into 'countries'...
+remote: Counting objects: 6, done.
+remote: Compressing objects: 100% (4/4), done.
+remote: Total 6 (delta 0), reused 6 (delta 0), pack-reused 0
+Unpacking objects: 100% (6/6), done.
+```
+
+Next we will set up a connection or **remote** to the authoritative repository 
+(the original version given to you by your instructor).
+In your browser, you can go back this repo by clicking on the link that says 'forked from'
+in the upper left hand corner, under your username and repo name.
+
+Copy the web address for this repo 
+(from the web address line or click the 'clone and download' and copy that).
+
+*TODO: Figure - Upstream Repo [../fig/github_screenshot_upstream_repo.png](https://github.com/UW-Madison-DataScience/git-novice-custom/tree/gh-pages/fig/github_screenshot_upstream_repo.png)*
+
+Then back in your terminal, navigate into the cloned repo and add the remote 
+connection to this repository.
+For this command we must give the remote a different nickname, 
+where our original remote is 'origin'
+this new remote will be called 'upstream'.
+You could give it a different nickname but 'upstream' is a common nickname for
+the authoritative repository.
+
+```shell
+$ cd countries
+$ git remote add upstream https://github.com/INSTRUCTOR-GIVEN/countries.git
+```
+
+> **If you tried copying the command above...**
+>
+> You will have to replace 'INSTRUCTOR-GIVEN' with the site your instructor
+> indicated at the beginning of this lesson. This will vary depending
+> on how your instructor set up for this lesson.
+
+At anytime you can see the remote connections your repo has using the following command:
+
+```shell
+$ git remote -v
+```
+
+```text
+origin	https://github.com/USERNAME/countries.git (fetch)
+origin	https://github.com/USERNAME/countries.git (push)
+upstream	https://github.com/INSTRUCTOR-GIVEN/countries.git (fetch)
+upstream	https://github.com/INSTRUCTOR-GIVEN/countries.git (push)
+```
+
+Now that we have this setup done we will be able to suggest 
```

Comment:
Also, somewhere I would like a cheatsheet (as shown in the branches section!) of the steps you take to set up a new PR: 
1. change to master, pull changes from upstream
2. update fork with master
3. create branch, commit changes
4. push branch
5. submit PR
6. make more changes, commit, push to update PR

Or something like that. Maybe at the end of this section, before conflicts?

Reply:
Good idea. I've added this to the very end of the pull request section.